miniconf/src/array.rs

```diff
@@ -101,32 +97,34 @@ macro_rules! depth {
             where
                 K: Keys,
             {
-                let item = get::<$y, N, _, _>(self, &mut keys)?;
+                let item = get(self, &mut keys, false)?;
                 item.get_by_key(keys).map_err(Traversal::increment)
             }
 
             fn get_mut_by_key<K>(&mut self, mut keys: K) -> Result<&mut dyn Any, Traversal>
             where
                 K: Keys,
             {
-                let item = get_mut::<$y, N, _, _>(self, &mut keys)?;
+                let item = get_mut(self, &mut keys, false)?;
                 item.get_mut_by_key(keys).map_err(Traversal::increment)
             }
         }
     )+}
 }
 depth!(2 3 4 5 6 7 8 9 10 11 12 13 14 15 16);
 
-// Y == 1
-impl<T, const N: usize> TreeKey for [T; N] {
+impl<const N: usize, T> TreeLookup for [T; N] {
```

Comment:
Why is the order of params swapped from below? I thought semantically const-generics had to come last?

Reply:
I don't think so. The impl order of const generics and types didn't seem to matter. And it shows up nowhere else.